In [2]:
import pandas as pd
from sqlalchemy import create_engine

In [52]:
green_tripdata = pd.read_csv("green_tripdata_2019-01.csv", parse_dates = ['lpep_pickup_datetime', 'lpep_dropoff_datetime'])
green_tripdata.columns = green_tripdata.columns.str.lower()
zones = pd.read_csv("taxi+_zone_lookup.csv")
zones.columns = zones.columns.str.lower()


In [55]:
engine = create_engine('postgresql://postgres:Miguel218159753@localhost:5432/ny_taxi')
engine.connect()

In [56]:
green_tripdata.to_sql(name = 'green_tripdata', index = False, con = engine, if_exists = 'replace')
zones.to_sql(name = 'zones', index = False, con = engine, if_exists = 'replace')

# Question 1. Knowing docker tags

Run the command docker --help build

Answer: --iidfile string

# Question 2. Understanding docker first run

Run the command docker run -it python:3.9 bash, and then 

pip list

Answer: There are 3 python packages in the python:3.9 image


# Question 3. Count records
How many taxi trips were totally made on January 15?

Tip: started and finished on 2019-01-15.

Remember that lpep_pickup_datetime and lpep_dropoff_datetime columns are in the format timestamp (date and hour+min+sec) and not in date.

In [58]:
query_3  = """
    SELECT COUNT(*)  
    FROM green_tripdata
    WHERE DATE(lpep_pickup_datetime) = '2019-01-15' 
           AND DATE(lpep_dropoff_datetime) = '2019-01-15';

"""
pd.read_sql(query_3, con = engine)

count
0  20530

# Answer:

20530

# Question 4. Largest trip for each day

Which was the day with the largest trip distance?

Use the pick up time for your calculations.

In [59]:
query_4  = """
    SELECT DATE(lpep_pickup_datetime) AS DAY, trip_distance AS Distance 
    FROM green_tripdata
    ORDER BY trip_distance DESC
    LIMIT 1;

"""
pd.read_sql(query_4, con = engine)

day  distance
0  2019-01-15    117.99

# Answer:

2019-01-15

# Question 5. The number of passengers

In 2019-01-01 how many trips had 2 and 3 passengers?

In [60]:
query_5  = """
    SELECT COUNT(*) 
    FROM green_tripdata
    WHERE DATE(lpep_pickup_datetime) = '2019-01-01'
    GROUP BY passenger_count;
    
"""
pd.read_sql(query_5, con = engine)

count
0     21
1  12415
2   1282
3    254
4    129
5    616
6    273

# Answer:

2: 1282 ; 3: 254

# Question 6. Largest tip

For the passengers picked up in the Astoria Zone which was the drop off zone that had the largest tip? We want the name of the zone, not the id.

Note: it's not a typo, it's tip , not trip

In [73]:
query_6  = """
    SELECT zones.zone, tip_amount 
    FROM green_tripdata INNER JOIN zones
            ON green_tripdata.dolocationid = zones.locationid
    WHERE pulocationid = (SELECT locationid FROM zones WHERE zone = 'Astoria')
    ORDER BY tip_amount DESC
    LIMIT 1;
    
"""
pd.read_sql(query_6, con = engine)

zone  tip_amount
0  Long Island City/Queens Plaza        88.0

# Answer:

Long Island City/Queens Plaza